src/tbp/monty/frameworks/models/evidence_matching/resampling_hypotheses_updater.py

```diff
@@ -399,69 +406,54 @@ def _sample_count(
+                beginning of the episode.
 
         Returns:
-            A tuple containing the number of existing and new hypotheses needed.
-            Existing hypotheses are maintained from existing ones while new hypotheses
-            will be initialized, informed by pose sensory information.
+            A tuple containing the hypotheses selection and count of new hypotheses
+            needed. Hypotheses selection are maintained from existing ones while new
+            hypotheses will be initialized, informed by pose sensory information.
 
         Notes:
-            This function takes into account the following ratios:
-              - `hypotheses_count_multiplier`: multiplier for total count calculation.
-              - `hypotheses_existing_to_new_ratio`: ratio between existing and new
-                hypotheses to be sampled.
+            This function takes into account the following parameters:
+              - `resampling_multiplier`: The number of hypotheses to resample. This
+                is defined as a multiplier of the number of nodes in the object graph.
+              - `evidence_slope_threshold`: This dictates how many hypotheses to
+                delete. Hypotheses below this threshold are deleted.
         """
         graph_num_points = self.graph_memory.get_locations_in_graph(
             graph_id, input_channel
         ).shape[0]
         num_hyps_per_node = self._num_hyps_per_node(channel_features)
-        full_informed_count = graph_num_points * num_hyps_per_node
 
-        # If hypothesis space does not exist, we initialize with informed hypotheses
-        if input_channel not in mapper.channels:
-            return 0, full_informed_count
+        # If hypothesis space does not exist, we initialize with informed hypotheses.
+        # Should we remove this now that we are resampling? We can sample the
+        # same number of hypotheses during initialization as in every other step.
+        if init_hyp_space:
+            full_informed_count = graph_num_points * num_hyps_per_node
+            return HypothesesSelection(maintain_mask=[]), full_informed_count
 
-        # Calculate the total number of hypotheses needed
-        current = mapper.channel_size(input_channel)
-        needed = current * self.hypotheses_count_multiplier
+        # This makes sure that we do not request more than the available number of
+        # informed hypotheses
+        resampling_multiplier = min(self.resampling_multiplier, num_hyps_per_node)
 
-        # Calculate how many existing and new hypotheses needed
-        existing_maintained, new_informed = (
-            needed * (1 - self.hypotheses_existing_to_new_ratio),
-            needed * self.hypotheses_existing_to_new_ratio,
-        )
+        # Calculate the total number of informed hypotheses to be resampled
+        new_informed = round(graph_num_points * resampling_multiplier)
+        new_informed -= new_informed % num_hyps_per_node
```

Comment:
Does this line just make sure `new_informed` is divisible by `num_hyps_per_node`?